drivers/sensor/lps22df/lps22df.c

```diff
@@ -0,0 +1,356 @@
+		 */
+		data->i3c_dev = i3c_device_find(cfg->i3c.bus, &cfg->i3c.dev_id);
+		if (data->i3c_dev == NULL) {
+			LOG_ERR("Cannot find I3C device descriptor");
+			return -ENODEV;
+		}
+	}
+#endif
+
+	if (lps22df_id_get(ctx, &id) < 0) {
+		LOG_ERR("%s: Not able to read dev id", dev->name);
+		return -EIO;
+	}
+
+	if (id.whoami != LPS22DF_ID) {
+		LOG_ERR("%s: Invalid chip ID 0x%02x", dev->name, id.whoami);
+		return -EIO;
+	}
+
+	LOG_DBG("%s: chip id 0x%x", dev->name, id.whoami);
+
+	/* Restore default configuration */
+	if (lps22df_init_set(ctx, LPS22DF_RESET) < 0) {
+		LOG_ERR("%s: Not able to reset device", dev->name);
+		return -EIO;
+	}
+
+	do {
+		if (!--tries) {
+			LOG_DBG("sw reset timed out");
+			return -ETIMEDOUT;
+		}
+		k_usleep(50);
+
+		if (lps22df_status_get(ctx, &status) < 0) {
+			return -EIO;
+		}
+	} while (status.sw_reset);
+
+	/* Set bdu and if_inc recommended for driver usage */
+	if (lps22df_init_set(ctx, LPS22DF_DRV_RDY) < 0) {
+		LOG_ERR("%s: Not able to set device to ready state", dev->name);
+		return -EIO;
+	}
+
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i3c)
+	if (cfg->i3c.bus != NULL) {
+		/*
+		 * Enabling I3C and disabling I2C are required
+		 * for I3C IBI to work, or else the sensor will not
+		 * send any IBIs.
+		 */
+
+		ret = lps22df_i3c_interface_set(ctx, LPS22DF_I3C_ENABLE);
+		if (ret < 0) {
+			LOG_ERR("Cannot enable I3C interface");
+			return ret;
+		}
+
+		ret = lps22df_i2c_interface_set(ctx, LPS22DF_I2C_DISABLE);
+		if (ret < 0) {
+			LOG_ERR("Cannot disable I2C interface");
+			return ret;
+		}
+	}
+#else
+	ARG_UNUSED(data);
+#endif
+
+	/* set sensor default odr */
+	LOG_DBG("%s: odr: %d", dev->name, cfg->odr);
+	ret = lps22df_set_odr_raw(dev, cfg->odr);
+	if (ret < 0) {
+		LOG_ERR("%s: Failed to set odr %d", dev->name, cfg->odr);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int lps22df_init(const struct device *dev)
+{
+	if (lps22df_init_chip(dev) < 0) {
+		LOG_DBG("Failed to initialize chip");
+		return -EIO;
+	}
+
+#ifdef CONFIG_LPS22DF_TRIGGER
+	if (lps22df_init_interrupt(dev) < 0) {
+		LOG_ERR("Failed to initialize interrupt.");
+		return -EIO;
+	}
+#endif
+
+	return 0;
+}
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "LPS22DF driver enabled without any devices"
+#endif
+
+/*
+ * Instantiation macros used when a device is on a SPI bus.
+ */
+
+#ifdef CONFIG_LPS22DF_TRIGGER
+#define LPS22DF_CFG_IRQ(inst) \
+	.gpio_int = GPIO_DT_SPEC_INST_GET(inst, drdy_gpios),
+#else
+#define LPS22DF_CFG_IRQ(inst)
+#endif /* CONFIG_LPS22DF_TRIGGER */
+
+#define LPS22DF_CONFIG_COMMON(inst)					\
+	.odr = DT_INST_PROP(inst, odr),					\
+	.lpf = DT_INST_PROP(inst, lpf),					\
+	.avg = DT_INST_PROP(inst, avg),					\
+	.drdy_pulsed = DT_INST_PROP(inst, drdy_pulsed),			\
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(inst, drdy_gpios),		\
+			(LPS22DF_CFG_IRQ(inst)), ())
+
+#define LPS22DF_SPI_OPERATION (SPI_WORD_SET(8) |			\
+				SPI_OP_MODE_MASTER |			\
+				SPI_MODE_CPOL |				\
+				SPI_MODE_CPHA)				\
```

Comment:
same, plus this could fit in half the lines if you want to keep it a bir mote compact

```c
#define LPS22DF_SPI_OPERATION (SPI_WORD_SET(8) | SPI_OP_MODE_MASTER |   \
                               SPI_MODE_CPOL | SPI_MODE_CPHA)           \
```